src/FieldWarning/Assets/Units/Component/Weapon/TargetingComponent.cs

```diff
@@ -95,11 +97,12 @@ public void WakeUp()
 
         private void Start()
         {
+            bool forExample 
             // TODO remove:
-            if (_weaponType == WeaponType.CANNON)
+            if (_weaponType is WeaponType.CANNON)
                 _weapon = new Cannon(
                         _data, _audioSource, _shotEffect, _shotSound, _shotVolume);
-            else if (_weaponType == WeaponType.HOWITZER)
+            else if (_weaponType is WeaponType.HOWITZER)
```

Comment:
I've seen Malum try to introduce this convention multiple times and I dislike it. Using `is` makes it seem like a type check. This case is particularly easy to talk about because the variable name makes the usage of `is` is extra misleading.

This is a normal comparison which when done with `==` in C# is obviously not a type check. If we do it with `is` we are using an operator with two meanings..